### DISCUSSION TITLE
Is it possible to use the same workflow in different repositories?

### DISCUSSION THREAD

**Discussion by raqyuste** [community/community#26327] [Actions]
<p>I have a lot repositories with the same architecture and very similar code and I would like to use the same workflow in each of them. There is some way to do this without duplicating a lot of yml files and not to have to modify all of them when I need to change the workflow?</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/raqyuste">@raqyuste</a> ,</p>
<p>As I mentioned in <a href="https://github.community/t/reference-a-workflow-file-from-a-different-repository/17715">this ticket</a>, the workflow files must be directly in the GitHub repository which can trigger the workflow runs. And currently we have no any workaround to allow a repository to reference workflow files from another repository.</p>

**Comment by mscoutermarsh** [comment, marked as answer]
<p>Not yet. It’s on our roadmap to handle this. Definitely important.</p>

**Comment by raqyuste** [comment]
<p>Okay. Many thanks, this would be very useful. </p>

**Comment by imduchy** [comment]
<p>Amazing! Definitely needed. </p>

**Comment by EricWF** [comment]
<p>Is your roadmap shared anywhere publicly?  </p>

**Comment by gwynne** [comment]
<p>This ability is desperately needed. Especially as workflows get more complex, custom actions are nowhere near powerful enough to counter the problem of workflow duplication. Are there any updates on when this might be available? Also, as has been asked in this thread already, is there any kind of public roadmap to give a sense of what the priorities are?</p>

**Comment by uday1kiran** [comment]
<p>Is this yet implemented?<br>
It is been 2 years but didn’t find any relevant links to this.<br>
Please guide me if it is already implemented</p>
